src/test/lambda/config/parameterUtils.test.ts

```diff
@@ -0,0 +1,237 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import {
+    getOverriddenParameters,
+    GetOverriddenParametersContext,
+    getParameterNames,
+    getParameters,
+    GetParametersContext,
+} from '../../../lambda/config/parameterUtils'
+
+describe('parameterUtils', async () => {
+    describe('getParameters', async () => {
+        it('returns an empty map if template has no parameters section', async () => {
+            const context: GetParametersContext = {
+                loadTemplate: async () => ({
+                })
+            }
+
+            const actual = await getParameters(vscode.Uri.file(''), context)
+            assert.strictEqual(actual.size, 0)
+        })
+
+        it('returns an empty map if parameters section is empty', async () => {
+            const context: GetParametersContext = {
+                loadTemplate: async () => ({
+                    Parameters: {}
+                })
+            }
+
+            const actual = await getParameters(vscode.Uri.file(''), context)
+            assert.strictEqual(actual.size, 0)
+        })
+
+        it('sets `required` to true if default is undefined', async () => {
```

Comment:
Yep, from [the docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html):

> Each parameter must be assigned a value at runtime for AWS CloudFormation to successfully provision the stack. You can optionally specify a default value for AWS CloudFormation to use unless another value is provided.

Reply:
If I'm reading that correctly, it says that all parameters must have a value, right? Does that mean that the next test/the code backing it is invalid (since you're making a parameter un-required)?

Reply:
In the next test, the value of `Default` is `false`. So even though it's not truthy, it's defined--and therefore, not required.